src/core/thread/mesh_forwarder.cpp

```diff
@@ -192,7 +192,7 @@ void MeshForwarder::ClearChildIndirectMessages(Child &aChild)
 
         message->ClearChildMask(mNetif.GetMle().GetChildIndex(aChild));
 
-        if (!message->IsChildPending())
+        if (!message->IsChildPending() && !message->GetDirectTransmission())
```

Comment:
Yes, but the impact is low.  This really only affects multicast messages (which may be sent using both direct and indirect transmissions).  Furthermore, it only occurs when a child is removed from the table after a multicast message is enqueued but before a direct transmission has been made.  Direct transmissions happen relatively quickly since they do not need to wait for data request commands.